component/core/src/main/java/org/exoplatform/social/core/listeners/FileAttachmentListener.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * This file is part of the Meeds project (https://meeds.io/).
+ * Copyright (C) 2023 Meeds Association contact@meeds.io
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.exoplatform.social.core.listeners;
+
+import org.exoplatform.commons.file.model.FileItem;
+import org.exoplatform.commons.file.services.FileService;
+import org.exoplatform.commons.file.services.FileStorageException;
+import org.exoplatform.services.listener.Event;
+import org.exoplatform.services.listener.Listener;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.social.metadata.attachment.AttachmentService;
+import org.exoplatform.social.metadata.model.MetadataItem;
+
+import java.io.IOException;
+import java.util.List;
+
+public class FileAttachmentListener extends Listener<Long, List<MetadataItem>> {
+
+  private static final Log LOG = ExoLogger.getLogger(FileAttachmentListener.class);
+
+  private FileService      fileService;
+
+  public FileAttachmentListener(FileService fileService) {
+    this.fileService = fileService;
+  }
+
+  @Override
+  public void onEvent(Event<Long, List<MetadataItem>> event) throws Exception {
```

Comment:
Please make this processing asynchronous since it's not needed to be blocker to Metadata items deletion.

```suggestion
@Asynchronous
public class FileAttachmentListener extends Listener<Long, List<MetadataItem>> {

  private static final Log LOG = ExoLogger.getLogger(FileAttachmentListener.class);

  private FileService      fileService;

  public FileAttachmentListener(FileService fileService) {
    this.fileService = fileService;
  }

  @Override
  @ExoTransational
  public void onEvent(Event<Long, List<MetadataItem>> event) throws Exception {
```